### PR TITLE
Add mvn configuration

### PR DESCRIPTION
**Pull request by bgzzz**
Add functionality to configure and export maven
options on the basis of encrypted yaml containing
needed credentials

**Description:**
Describe your changes.

**Related issue:**
Add link to the related issue.

**Check list:**
- [ ] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

### CODE REVIEW

**Review by blacksails — COMMENTED** [review]
What if your project is built using gradle? Then this action is a bit weird to use.

File: src/auth.ts [hunk 1]
```diff
@@ -85,3 +91,56 @@ async function write(directory: string, settings: string) {
...
+  );
+
+  fs.writeFileSync(
+    path.join(settingsDir, 'settings-security.xml'),
+    btoa(creds.MVN_SECURITY_SETTINGS),
+    {
+      encoding: 'utf-8',
+      flag: 'w'
+    }
+  );
```
Comment (blacksails) [inline src/auth.ts:125]: Do we need to overwrite files. Wouldn't it be better and more transparent to explicitly set the settings that we need in the case that the file already exists?

**Comment by bgzzz** [comment]
> What if your project is built using gradle? Then this action is a bit weird to use.

it can be used with gradle. i can make a parameter optional, that is not a problem

File: src/auth.ts (hunk 1 above)
Comment (bgzzz) [reply src/auth.ts:125]: i did not get. can you elaborate here. i assume it should also work if files not exist

Comment (blacksails) [reply src/auth.ts:125]: It is a hypothetical, but say the files does exist before running this, wouldn't you overwrite what is there already?

**Review by blacksails — APPROVED** [review]

**Pull request closed without merging**